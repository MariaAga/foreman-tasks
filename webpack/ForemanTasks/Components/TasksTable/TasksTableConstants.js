import { getControllerSearchProps } from 'foremanReact/constants';

export const TASKS_TABLE_CONTROLLER = 'tasks';
export const TASKS_TABLE_SUCCESS = 'TASKS_TABLE_SUCCESS';
export const TASKS_TABLE_PENDING = 'TASKS_TABLE_PENDING';

export const TASKS_SEARCH_PROPS = getControllerSearchProps('tasks');
