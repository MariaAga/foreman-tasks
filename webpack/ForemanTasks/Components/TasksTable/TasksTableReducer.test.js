import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import { TASKS_TABLE_SUCCESS } from './TasksTableConstants';
import reducer from './TasksTableReducer';

const fixtures = {
  'should return the initial state': {},
  'should handle TASKS_TABLE_SUCCESS': {
    action: {
      type: TASKS_TABLE_SUCCESS,
      payload: {
        subtotal: 120,
        page: 3,
        per_page: 10,
      },
    },
  },
};

describe('TasksTablePaginationReducer reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
