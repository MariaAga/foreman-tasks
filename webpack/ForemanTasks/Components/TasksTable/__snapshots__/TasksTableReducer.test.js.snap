// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksTablePaginationReducer reducer should handle TASKS_TABLE_SUCCESS 1`] = `
Object {
  "tasksTablePagination": Object {
    "itemCount": 120,
    "loading": false,
    "pagination": Object {
      "page": 3,
      "perPage": 10,
    },
  },
}
`;

exports[`TasksTablePaginationReducer reducer should return the initial state 1`] = `
Object {
  "tasksTablePagination": Object {
    "itemCount": 0,
    "loading": true,
    "pagination": Object {
      "page": 1,
      "perPage": 10,
    },
  },
}
`;
